flow/data_pipeline/data_pipeline.py

```diff
@@ -0,0 +1,284 @@
+"""contains class and helper functions for the data pipeline."""
+import pandas as pd
+import numpy as np
+import boto3
+from flow.data_pipeline.query import QueryStrings, testing_functions
+from time import time
+
+
+def generate_trajectory_table(data_path, extra_info, partition_name):
+    """Generate desired output for the trajectory_table based on standard SUMO emission.
+
+    Parameters
+    ----------
+    data_path : str
+        path to the standard SUMO emission
+    extra_info: dict
+        extra information needed in the trajectory table, collected from flow
+    partition_name: str
+        the name of the partition to put this output to
+    Returns
+    -------
+    output_file_path: str
+        the local path of the outputted csv file
+    """
+    raw_output = pd.read_csv(data_path, index_col=["time", "id"])
+    required_cols = {"time", "id", "speed", "x", "y"}
+    raw_output = raw_output.drop(set(raw_output.columns) - required_cols, axis=1)
+
+    extra_info = pd.DataFrame.from_dict(extra_info)
+    extra_info.set_index(["time", "id"])
+    raw_output = raw_output.merge(extra_info, how="left", left_on=["time", "id"], right_on=["time", "id"])
+
+    # add the partition column
+    # raw_output['partition'] = partition_name
+    raw_output = raw_output.sort_values(by=["time", "id"])
+    output_file_path = data_path[:-4]+"_trajectory.csv"
+    raw_output.to_csv(output_file_path, index=False)
+    return output_file_path
+
+
+def generate_trajectory_from_flow(data_path, extra_info, partition_name):
```

Comment:
Not sure I understand what this method is doing, looks like it takes a dataframe and saves it to two files (one without a header). Does 'extra_info' contain a trajectory object?

Reply:
`extra_info` contains all the information we want in the trajectory table. Sorry for the confusing variable names, it used to be information we need "extra" from the SUMU output. All this method does is save it as CSV files. The one without header is uploaded to S3 and deleted locally later. Headers cause issue when Athena tries to interpret the CSV file. The one with header is stored locally.